core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -129,21 +129,11 @@ private[spark] object BlockManagerId {
   def apply(in: ObjectInput): BlockManagerId = {
     val obj = new BlockManagerId()
     obj.readExternal(in)
-    getCachedBlockManagerId(obj)
```

Comment:
A reminder for myself: cache is still needed for the use case in `UpdateBlockInfo`.